solver.md

```diff
@@ -0,0 +1,461 @@
+to all solver components. For the most part, these parameters dictate
+things like stopping conditions, which apply to all of the solver
+components that will be used.
+
+Following the general parameters are one or more blocks that specify
+solver components, which direct the solver to run particular solution
+algorithms. For example:  
+
+```
+       <broyden-solver-component>
+           <max-iterations>25</max-iterations>
+           <ftol>1.0e-3</ftol>
+           <solution-info-filter>solvable-nr || (market-type="Tax" &amp;&amp; solvable)</solution-info-filter>
+       </broyden-solver-component>
+```
+
+The first few lines specify parameters that are specific to this
+particular solver component. The `solution-info-filter` line gives a
+[predicate](#Solution-info-filter-Predicates) for determining which
+markets the component will attempt to solve. This filter allows us to
+exclude from a solver component markets that are likely to cause that
+component to fail (for example, markets that make the Jacobian matrix
+singular for the Broyden solver) or whose solution is unlikely
+to be improved by the component.
+
+The algorithms will be run consecutively in the order listed in the
+configuration file. When all have been run, the termination criteria
+will be checked. The sequence of components will be repeated until
+either the model is solved or the maximum number of model evaluations
+has been exceeded.
+
+<a name="General-Solver-Parameters"></a>
+
+### General Solver Parameters
+
+* **calibration-tolerance**: During the calibration periods, the
+  tolerance for determining whether or not the calibration was
+  successful.
+* **max-model-calcs**: The maximum number of times the full model will
+  be evaluated in the course of the solution. If the solver has not
+  found a valid solution by that time, it exits and issues a warning.  
+* **solution-tolerance**: The relative excess demand threshold, below
+  which the model is considered solved. Loosely speaking, this is the
+  accuracy to which we wish to solve the model.  
+* **solution-floor**: The absolute excess demand threshold, below
+  which the model is considered solved. This parameter covers the
+  cases where a low demand value makes achieving a relative excess
+  demand threshold unfeasible.  
+
+<a name="Solution-info-filter-Predicates"></a>
+
+### Solution-info-filter Predicates
+
+The predicates available to filter markets are:
+
+* **all**: True for any market.  
+* **market-name**: True if the name of the market matches the name
+  given in the filter.  
+* **market-type**: True if the type of the market matches the type
+  given in the filter. The available market types are Normal,
+  Calibration, Inverse-Calibration, Tax, RES, Subsidy,
+  [Trial-Value, Demand, and Price](Cycle-breaking-in-GCAM.html).  
+* **solvable**: True for markets that should be solved. "Solvable" in
+  this context refers to intent, not capability. A market is
```

Comment:
Agreed, but that's the terminology that GCAM has been using since before I got here.